### PR TITLE
improved Account so that all tests in AccountTest pass

### PR DESCRIPTION
**Pull request by RF-Buerky**
Solved Issue #21

### CODE REVIEW

**Comment by RF-Buerky** [comment]
The test failure has occured because all the tests are executed. So all the eleven tests in
org.codefx.demo.bingen.energy_model.math.RomanNumeralsTest
failed.

View the tests log lines 967 and 968 to get the proof. There you can read:
"Running org.codefx.demo.bingen.energy_model.AccountTest
Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.001 sec"

File: src/main/kotlin/org/codefx/demo/bingen/energy_model/Account.kt [hunk 1]
```diff
@@ -21,5 +29,59 @@ class Account(var balance: Int, var limit: Int) {
...
+
+    // Referring to the function "negativeDepositDoesNotChangeBalance()", there should be no
+    // incident on the balance if a negative amount is about to get deposit. This is a plausible
+    // way as it is not possible to deposit a negative amount of money onto your bank account.
+    // The equivalent way would be to withdraw the (positive) amount of money. So this is the way
+    // we are going here, also.
+    fun deposit(amount: Int): Int {
+        var deposited_amount : Int = 0
+        if (amount < 0){
+            deposited_amount = 0
```
Comment (nicolaiparlog) [inline src/main/kotlin/org/codefx/demo/bingen/energy_model/Account.kt]: In Java and Kotlin it is common to write variables in CamelCase, e.g. `depositedAmount`.

File: src/main/kotlin/org/codefx/demo/bingen/energy_model/Account.kt [hunk 2]
```diff
@@ -21,5 +29,59 @@ class Account(var balance: Int, var limit: Int) {
...
+        var withdrawed_amount : Int = 0
+
+        if (amount < 0) {
+            withdrawed_amount = 0
+            balance = balance
+        } else if (balance >= amount) {
+                // balance suffices, pay out the money
+            withdrawed_amount = amount
+            balance = balance - amount
+        } else if (balance < amount && balance-amount>limit) {
```
Comment (nicolaiparlog) [inline src/main/kotlin/org/codefx/demo/bingen/energy_model/Account.kt]: This branch does exactly the same as the one before, which means you can "merge" them, getting:

```
else if (balance >= amount || balance < amount && balance-amount>limit) {
                // balance suffices, pay out the money
            withdrawed_amount = amount
            balance = balance - amount
}
```

Taking a closer look, you can simplify the condition to just `balance-amount > limit`, which makes sense. Money can be withdrawn when what you have minus what you want is above your limit.

**Comment by RF-Buerky** [comment]
Just like before. All the tests for `Account` pass. The failure is a result of the other tests for `RomanNumeralsTest`.

File: src/main/kotlin/org/codefx/demo/bingen/energy_model/Account.kt (hunk 1 above)
Comment (RF-Buerky) [inline src/main/kotlin/org/codefx/demo/bingen/energy_model/Account.kt]: ok, change has been done

File: src/main/kotlin/org/codefx/demo/bingen/energy_model/Account.kt (hunk 2 above)
Comment (RF-Buerky) [inline src/main/kotlin/org/codefx/demo/bingen/energy_model/Account.kt]: oh yes thanks, changes have been done.

Comment (RF-Buerky) [inline src/main/kotlin/org/codefx/demo/bingen/energy_model/Account.kt]: brilliant, thanks. the changes have been done

**Comment by RF-Buerky** [comment]
Just like before. All the tests for `Account` pass. The failure is a result of the other tests for `RomanNumeralsTest`.

**Comment by nicolaiparlog** [comment]
Homework solved. :+1:

**Pull request closed without merging**